### ISSUE TITLE
Can Keras use processed output tensor rather than just output tensort to compute the loss tensor?

### ISSUE THREAD

**Issue by speedcell4** [fchollet/keras#4604]
In Tensorflow, I can do something like this

```python
x = tf.placeholder(tf.float32, shape=[None, input_dim], name='x')
y_true = tf.placeholder(tf.int64, shape=[None], name='y_true')

y_true_one_hot = tf.one_hot(y_true, output_dim, on_value=1.0, off_value=0.0)

h1 = fully_connected(x, input_dim, activation_fn=tf.nn.sigmoid)
h2 = fully_connected(h1, input_dim, activation_fn=tf.nn.sigmoid)
y_pred_logits = fully_connected(h2, output_dim, activation_fn=tf.nn.softmax)

y_pred = tf.argmax(y_pred_logits, axis=1)

loss = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(y_pred_logits, y_true_one_hot))
optimizer = tf.train.AdamOptimizer(0.01).minimize(loss)
```

Please notice the `loss` is computed by `y_pred_logits`, instead of just using `y_pred`, while I still use `y_pred`, as the output of the model, to predicate.

Can Keras do the same thing? I mean, seems the Keras `Model.compile`'s parameter `optimizer` automatically just use the output of `Model`...

**Comment by jmhessel** [comment]
This isn't really a bug, so I would post things like this in the [google group](https://groups.google.com/forum/#!forum/keras-users) in the future.

Yes, this is possible. The way I do it is to have the keras model output the loss, and have the target be an appropriately-sized tensor of zeros. The loss function can then be something like mean absolute error, or you can make your own lambda expression.
